pkg/cli/rm_helper.go

```diff
@@ -150,14 +152,36 @@ func removeApp(arg string, c client.Client, cmd *cobra.Command, force bool) erro
 		}
 	}
 	app, err := c.AppDelete(cmd.Context(), arg)
-
 	if err != nil {
 		return fmt.Errorf("deleting app %s: %w", arg, err)
 	}
-	if app != nil {
-		fmt.Println("Removed: " + arg)
-	} else {
+
+	if app == nil {
 		fmt.Printf("Error: No such app: %s\n", arg)
+		return nil
+	}
+
+	// Ensure this gets printed whether we wait or not
+	defer func() {
+		if err == nil || apierrors.IsNotFound(err) {
+			fmt.Println("Removed: " + arg)
+		}
+	}()
+	if wait {
+		fmt.Printf("Waiting for app %s to be removed...\n", arg)
+		for {
+			select {
+			case <-cmd.Context().Done():
+				return cmd.Context().Err()
+			default:
+				if _, err = c.AppGet(cmd.Context(), arg); apierrors.IsNotFound(err) {
+					return nil
+				} else if err != nil {
+					logrus.Debugf("Error getting app for removal check: %v", err)
```

Comment:
@thedadams and I talked about this on Slack. Here's the discussion for posterity:

>  We are trying to abstract Kubernetes away as much as possible, so printing the error seemed counter to that goal.

That's a great point.

I think acorn should probably say something if it encounters an error here. For the short term, I feel like that could just be some boilerplate "command failed" message; but long-term, we'll probably want to give the user a more useful message.

Zooming out at the big picture, for any `$PLATFORM` that `acorn` supports -- here `PLATFORM=k8s` -- there are two primary approaches that jump out at me:
A) Maintain mappings between user-facing `acorn` error types and `$PLATFORM` error types
B) Be transparent with `$PLATFORM` error types

and besides the obvious toil on our end, my intuition is that there are cases where `A` isn't possible without leaking `$PLATFORM` specific details into `acorn` error types anyway.

I think `B` still holds up the abstraction goal as long as all platforms get the same treatment; i.e. none of their errors becomes codified in `acorn`'s top-level error types.

so I'd probably opt for `B`, but hide the output behind the --debug flag.